### ISSUE TITLE
HREF attribute stripped down from Privacy or Cookie content text

### ISSUE THREAD

**Issue by marcocevoli** [alextselegidis/easyappointments#719]
Hi,
if you insert a link into the Content section under Privacy or Cookies or Terms and Conditions, the HREF attribute part is stripped down (delete) when you abandon the page. I was wondering if there is any security mechanism that I can disable, so that URLs can be inserted (and saved).

EDIT: I've pointed down the issue to this: https://stackoverflow.com/questions/45913044/ckeditor-automatically-removes-style-attribute-and-add-xss-attribute-removed but I don't know what I can do to solve it.

**Comment by yrammos** [comment]
@marcocevoli I was also bogged down by CodeIgniter's heavy-handed HTML sanitization in [my fork](https://github.com/tekhnee/appointments). In my case it was the service descriptions that I needed less aggressively sanitized. Eventually, I disabled CodeIgniter's sanitization facilities altogether and replaced them with [HTML Purifier](http://htmlpurifier.org/). To outline the steps, one needs need to:

1. Disable CodeIgniter's sanitization (`$config['global_xss_filtering'] = FALSE;
` in `/src/application/config/config.php`).
2. Include the HTML purifier helper in the CodeIgniter project (see docs).
3. Declare one's own sanitization preferences in `/src/application/helpers/htmlpurifier_helper.php` lines 34-43.
4. Find the relevant back-end controller (which, in your case, passes the Privacy/Cookies/T&C content from the form to the database) and filter that data through HTML Purifier. This will likely involve something along the lines of:

```php
   $post_data = $this->input->post('post_data');   // Existent line.
   $this->load->helper('htmlpurifier');    // New line
   $post_data = html_purify($post_data_dirty);  // New line
```

**Comment by alextselegidis** [comment]
Hello! 

This is fixed in v1.4.0

<small>
<table>
<tr>
<td><img src="https://easyappointments.org/img/easyappointments-signature.png"></td>
<td>&nbsp;&nbsp;Alex Tselegidis, Easy!Appointments Creator <br>
&nbsp;&nbsp;<a href="https://alextselegidis.com">Need a customization? Contact me in person!</a></td>
</tr>
</table>
</small>
